lambda/lambda.mli

```diff
@@ -762,9 +762,31 @@ and lambda_event_kind =
   | Lev_function
   | Lev_pseudo
 
+(* Descriptor for a parameter that this module takes at runtime. *)
+type runtime_param_descr =
+  | Rp_argument_block of Global_module.t  (* The argument block of a module
+                                             compiled with [-as-argument-for] *)
+  | Rp_dependency of Global_module.t      (* A parameterised module (not itself a
+                                             parameter) that this module depends
+                                             on *)
+  | Rp_unit                               (* The unit value (only used when
+                                             there are no other parameters) *)
+
+type module_block_format =
+  | Mb_record of { mb_size : int }      (* A block with [mb_size] fields *)
```

Comment:
Maybe just `Mb_normal` and `Mb_parameterized`.  I feel "record" carries too much connotation (and also this doesn't have field names, so not really record)

Reply:
I feel strongly that this is the place where we've passed through the nice typechecker view of the world and into the middle end, and these types should reflect that: we're talking about what the actual lambda terms consist of, not what was going on in the modules. I've added a comment describing the correspondence, though. I see your point about the word _record_, but I don't see a better word than _block_ (and it doesn't help much to say the block is a block). _Normal_ carries no information. Maybe `Mb_struct`? (Then the other might be `Mb_wrapped_functor`.)

Reply:
I take your point that lambda is middle-end and the names should sound so. However, given that the type name still involves "module", I think we can still mention "parameterized".

"Normal" indicates that it's the common case - note that `cmi_format.kind` already have two constructors "Normal" and "Parameter".

`Mb_struct` combined with the name "module" give the impression that the fields correspond to values in a `struct .. end` (is that true?)

Reply:
I wouldn't mind a better name in `Cmi_format.kind` as well, if we can think of a good retronym for “module that's not a parameter module.” Tempting to say _static module_ but that's not quite right: a parameterized module isn't quite static (it varies by what arguments you give it). The salient property is that the module has one specific definition (possibly an open term), but _defined module_ seems weird.

Yeah, the fields will be exactly the values as they would be in a `struct .. end` with an inferred signature (there are sometimes also a few extra fields, like the argument block, but the basic picture of “stuff stored like a `struct`” is accurate).

Reply:
Ah, maybe I should think of a better name there. The issue is that actually it's _always_ a struct (changing that expectation might be possible but seems more disruptive than necessary), so in the case that we want to store a functor, we instead have to make a struct with just the functor in it (stuffing it in a one-element struct is what I meant by “wrapped”). Maybe I should just call it `Mb_functor` but I'd prefer that `lambda.mli` tell as few lies as possible, if we can manage